core/app/models/spree/ability.rb

```diff
@@ -64,6 +63,9 @@ def initialize(user)
         ability = clazz.send(:new, user)
         @rules = rules + ability.send(:rules)
       end
+
+      # Protect admin and user roles
+      cannot [:update, :destroy], Role, name: ['admin', 'user']
```

Comment:
I think admin is all we need to protect.  The user role isn't actually used anywhere as far as I know.